### DISCUSSION TITLE
Make it possible to archive/deprecate a parser

### DISCUSSION THREAD

**Discussion by madsnedergaard** [electricitymaps/electricitymaps-contrib#3473] [Ideas] [answered]
### Description

Currently, all parsers are running even though we know some of them will never get data. 

#3337 (AM) is a good example, where the data source is no longer available.
The parser is still there and running, so the map displays data as "temporarily unavailable":
![Screenshot 2021-11-05 at 11 28 01](https://user-images.githubusercontent.com/3296643/140496812-91f5ba55-f57e-4a58-ad72-268d92a9abea.png)

It would be nice if we could ensure that these parsers don't run and make it clear to everyone that data is not available. We could consider just deleting the code, but I think there's value in keep the old parsers in case the data becomes available again on a new URL or at least to make history more clear for new contributors (having to go through git history to find this is not great).

### Proposed solution

- Create a new folder in `/parsers` called `archive`, `deprecated` or something similar
- Test all parsers and move all that will never work (maybe those that have been down for a month or more, see also["known parser issues"](https://github.com/tmrowco/electricitymap-contrib/wiki/Parser-issues))
- Ensure that parsers in that folder are ignored when the script runs 

Is there anything I'm missing or any good reasons for **not** doing this?

**Comment by madsnedergaard** [comment]
@FelixDQ, @skovhus and @corradio, do you have any inputs to this? I remember there's previously been discussions around this, but I couldn't find anything on Github :)

**Comment by byronwilliams** [comment]
I think the parsers should remain where they are inside the parsers folder. The zones.json file should have a property added called "disabled" and "disabledReason". 

If a zone is disabled the map should have some wording saying something like "Know of a more up to date data source? Let us know"

**Reply by madsnedergaard** [reply]
What's your thoughts behind keeping them in the same folder vs having a subfolder for deprecated parsers? I feel that the parser's code will be very detached and you have to lookup a zone in zones.json in order to know if the parser is actually used :)

Also, I was thinking it's easier for new contributors to join in if they see a folder with deprecated parsers that they can try to get working again, but maybe that's just me?

**Reply by VIKTORVAV99** [reply]
I believe it makes sense to move them to a specific folder for a few reasons.
1. It's easier to spot a active vs archived parser.
2. We really don't want to include more information than absolutely necessary in the zones.json file as it's currently bundled with the frontend. (it's already a problem)
3. Keeping it in a separate folder will make it easier to use with CI pipelines in the future if we so wish. (Like checking if data is available again or labeling pr's that target a archived parser)

**Reply by madsnedergaard** [reply]
For the reasons listed above I think we should roll with separate folder! If anyone wants to pick it up that would be amazing (otherwise I'll add it to the backlog) :)

**Comment by skovhus** [comment]
I also think we should keep the parser code around, as someone might want to fix, improve or use it as inspiration.

Moving the code is one option, another is annotating the parser as @byronwilliams suggested. I have no strong preference.

**Comment by q--** [comment]
If parsers are really dead, one option would be to remove them and list them in a file similar to DATA_SOURCES.md with a link to the last commit before they were removed. (Because sometimes a data source re-appears, as I know happened with Iraq - if under a different URL.)

For example:

> # Removed parsers
>The data source for these parsers has been gone for more than (e.g.) a year and it appears unlikely to come back online. If it does, you can find the last version of the parser code here.
>
>| Code | Zone Name | Last version of parser before removal |
>| - | - | - |
>| AM | Armenia | [`AM.py`](https://github.com/electricitymap/electricitymap-contrib/blob/f0b735dc7878239cdbff03f4976f906c066eeb8c/parsers/AM.py) |
>

**Reply by madsnedergaard** [reply]
Good suggestion, for now we've gone with the `/parsers/archived/` directory approach instead. If that becomes too much we should consider this :)

**Comment by madsnedergaard** [comment, marked as answer]
Fixed in https://github.com/electricitymap/electricitymap-contrib/pull/4163 :)
